### PR TITLE
Mmi 5414 user reported error message in de fi data

### PR DESCRIPTION
**Pull request by jpsains**
* Tidied up the config object 
* Introduced new ChainName const so that I could apply stricter type checking (couldnt find a way to type check config object without it and now the IConfig correctly exposes object keyed by chain name)
* renamed old ChainName type to ChainIdToChainNameMap

### CODE REVIEW

File: packages/adapters-library/src/config.ts [hunk 1]
```diff
@@ -102,90 +132,33 @@ const ConfigSchema = z
...
+        [ChainNames.bsc]: z.number(),
+        [ChainNames.matic]: z.number(),
+        [ChainNames.ftm]: z.number(),
+        [ChainNames.base]: z.number(),
+        [ChainNames.arb]: z.number(),
+        [ChainNames.avax]: z.number(),
+        [ChainNames.linea]: z.number(),
       })
-      .default({}),
+      .default(defaultMaxBatchSize),
```
[the remark this answers is not in the document]
Comment (jpsains) [reply packages/adapters-library/src/config.ts]: image

image

[the remark this answers is not in the document]
Comment (jpsains) [reply packages/adapters-library/src/config.ts]: yes you can

[the remark this answers is not in the document]
Comment (jpsains) [reply packages/adapters-library/src/config.ts]: Just checking behaviour when passing an object

[the remark this answers is not in the document]
Comment (jpsains) [reply packages/adapters-library/src/config.ts]: image
This throws an error

[the remark this answers is not in the document]
Comment (jpsains) [reply packages/adapters-library/src/config.ts]: image
I dont think its bad, is it? Change the defaults and expect pain?

[the remark this answers is not in the document]
Comment (bergarces) [reply packages/adapters-library/src/config.ts]: If you set the default at entry level, you can just overwrite one chain and let the others use the default.

If you set the default at object level, then you need to overwrite all chains or none.

Unless there's some magic way of merging whatever you've passed and the default object.

[the remark this answers is not in the document]
Comment (jpsains) [reply packages/adapters-library/src/config.ts]: so you can change individual at env var level but not object level

[the remark this answers is not in the document]
Comment (bergarces) [reply packages/adapters-library/src/config.ts]: The problem is that what zod says now is:
"If you don't pass a `maxBatchSize`, here is a default. But if you pass `maxBatchSize`, it needs to be a complete object that redefines every single chain".

Before it said:
"If you pass a partially filled `maxBatchSize`, which includes not passing anything at all, each field inside will be filled with the default for that field".

[the remark this answers is not in the document]
Comment (bergarces) [reply packages/adapters-library/src/config.ts]: I think you could, however, achieve a similar behaviour as we had before with `preprocess`.

```
maxBatchSize: z
    .preprocess((input) => ({
      ...maxBatchSize,
      ...input,
    }),
      z.object({
      [ChainName.ethereum]: z.number(),
      [ChainName.op]: z.number(),
      [ChainName.bsc]: z.number(),
      [ChainName.matic]: z.number(),
      [ChainName.ftm]: z.number(),
      [ChainName.base]: z.number(),
      [ChainName.arb]: z.number(),
      [ChainName.avax]: z.number(),
      [ChainName.linea]: z.number(),
    }))
    .default(maxBatchSize),
```

[Zod Playground](https://zod-playground.vercel.app?appdata=N4IgzgxgFgpgtgQxALhALwHQHsBGArGCAFwApgAdAOwAJqAHBAJyIEsEAbAeX0KIHUWRKADEWMdgBMAIjABmCAK7siYZNUy4CxMlVq1ZYyQEE1mMEUYtKAcxIBKDBLmLlJAOQHxEo27sAaXT1PSQAhUwxzSxt7R2clUg9DCRDfQIBfByd5eLIMgJpqTV4BIRls5XCi7QoC-SSTdQiLK1t-QLqvMMbIlvt0zLjXYGpg7zVErx8-EaSuidC3ajzAhmY2Lh5iEpEkspcVAFUwFoAVRgRKMFksRjhwukYYB6wIGDAwEnbqEis6BSI7NQALwAPm%2BNT0egw0OGowa828bmmozmKMWaXykNo0Iwv3%2BmL0y1qjSqpAhWLh4R60TaxI6oSpzRpBNoGUCA3KZKJGRAfhAADcOAo3igANogcm0VasDjcLT8QQ7Lx7eJgI6nc6Xa63NSSin1cZGI1uL5pKhpEAAXT5-JgjGOWEoKBAAGYMAAmN0ADhAaSAA)

**Pull request merged**